src/Microsoft.AspNet.Server.Kestrel/Http/Listener.cs

```diff
@@ -13,8 +13,10 @@ namespace Microsoft.AspNet.Server.Kestrel.Http
     /// </summary>
     public abstract class Listener : ListenerContext, IDisposable
     {
-        protected Listener(ServiceContext serviceContext) : base(serviceContext)
+        protected Listener(ServiceContext serviceContext) 
+            : base(serviceContext)
         {
+            Memory2 = new MemoryPool2();
```

Comment:
The byte/char pool is used for the text writer. All signs point to that thing's strategy being changed for performance. I don't think the original IMemoryPool and MemoryPool will still be around after that.